lib/net/ssh/config.rb

```diff
@@ -346,13 +351,17 @@ def eval_match_conditions(condition, host, settings)
               exprs.split(",").each do |expr|
                 condition_met = condition_met || host =~ pattern2regex(expr)
               end
-              matching = matching && negated ^ condition_met
+              matching = true && negated ^ condition_met
```

Comment:
I don't think it's correct, when there are multiple condiitons, for example:

```ruby
Match host *.home.lan host !foo.home.lan
```

I think your code will condsider google.com as match.

Reply:
Indeed, that was missed. Have updated logic to correctly handle this situation and added relevant test.